src/Sylius/Bundle/CoreBundle/StateMachine/CompositeStateMachine.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\StateMachine;
+
+use Exception;
+use Traversable;
+use Webmozart\Assert\Assert;
+
+class CompositeStateMachine implements StateMachineInterface
+{
+    /** @var array<StateMachineInterface> */
+    private array $stateMachineAdapters;
```

Comment:
It can be manipulated through tag priorities, but I can also make a configuration which will just swap their priorities.

Reply:
I was hoping for something more explicit and possibly per SM definition, for example `sylius_order` via `winzou` and `sylius_shipment` via `workflow`

Example scenario,
Existing end-user app has complex winzou based usage, extra transitions, states, guards, callbacks, number of App specific SMs,
End-user decides to give it a try to new sylius workflow SMs and would like to implement new feature based of `sylius_shipment` SM, but without porting all existing sylius SMs related functionality first.

Most likely I'm over complicating it or missing the point, for example how `Core\StateResolvers` will work in such case, etc.
It would be good to see usage example based on this abstraction.

Reply:
I'll try something like
```
sylius_core:
	state_machine:
		adapter:
			default: 'symfony_workflow' # or 'winzou_state_machine'
			graph:
				sylius_shipment: 'winzou_state_machine'
				
```

Reply:
Done :)